client/js/analysis.js

```diff
@@ -1,25 +1,113 @@
+
 window.onload = function() {
-  for (let i = 1; i < 6; i++) {
+  // Listen for submission click
+  const formSubmit = document.getElementById('sendbutton');
+  formSubmit.addEventListener('click', fetchResponse, false);
+};
+
+/**
+ * Fetches the data from the backend and populates the screen
+ *
+ */
+async function fetchResponse() {
+  const videoQuery = document.getElementById('searchbar').value;
+  const response = await fetch(`/api/analysis?name=${videoQuery}`);
+  const responseJson = await response.json();
+  populationHandler(responseJson);
+}
+
+/**
+ * Handles populating all elements on the screen
+ *
+ * @param {object} videoAnalysis the object returned from the HTTP request
+ * which contains all of the data.
+ */
+function populationHandler(videoAnalysis) {
+  const charts = document.getElementById('charts');
+  const list = document.getElementById('list');
+
+  removeAllChildNodes(charts);
+  removeAllChildNodes(list);
+  populateDonutCharts(videoAnalysis.perspectiveMap);
+  // The +1 allows for determineOverall() to run correctly.
+  const time = renderComments(videoAnalysis.commentArray) + 1;
+  const overall = determineOverall(videoAnalysis.magnitudeAndScore.magnitude,
+      videoAnalysis.magnitudeAndScore.score);
+  setTimeout(() => {
+    addFeedbackResult(overall);
+  }, time*COMMENT_APPEARANCE_TIME);
+}
+
+/**
+ * Puts the charts on the screen
+ *
+ * @param {Map<string, number>} map the map from the videoAnalysis object
+ */
+function populateDonutCharts(map) {
+  for (const key in map) {
+    if (Object.prototype.hasOwnProperty.call(map, key)) {
+      // Convert to string for type compatibility
+      addDonutChart(key, map[key]);
+    }
+  }
+}
+
+/**
+ * Puts the comments on the screen
+ *
+ * @param {Array} array the comment array from the json response
+ * @returns {number} the stop index so that overall knows when to come in
+ */
+function renderComments(array) {
+  const totalComments = Math.min(COMMENT_TO_STOP_AT, array.length);
+  for (let i = 0; i < totalComments; i++) {
+    const filteredValue = array[i].replace('\n', '');
     setTimeout(() => {
-      addListElement(i);
-    }, i * 1500);
+      addListElement(filteredValue);
+    }, (i+1) * COMMENT_APPEARANCE_TIME);
   }
-  setTimeout(() => {
-    addFeedbackResult('POSITIVE');
-  }, 9000);
-
-  for (let i = 0; i < 6; i++) {
-    const values = [
-      'TOXICITY',
-      'IDENTITY ATTACK',
-      'INSULT',
-      'PROFANITY',
-      'THREAT',
-      'FLIRTATION',
-    ];
-    addDonutChart(values[i], Math.floor(Math.random() * 61) + 30);
+  // so that the overall knows when to come in
+  return totalComments;
+}
+
+/**
+ * Determines the overall sentiment and adds it to the screen
+ *
+ * @param {number} magnitude the magnitude returned by Perspective API
+ * @param {number} score the score returned by Perspective API
+ * @returns {string} a string describing the magnitude and score in words
+ */
+function determineOverall(magnitude, score) {
```

Comment:
Consider renaming this to 'determineOverallSentiment' or just 'determineSentiment'. While reading through this PR, it felt confusing seeing "overall" referred to because it leaves the question "overall what?" unanswered.